src/storage/repository_plugins/sql/__init__.py

```diff
@@ -0,0 +1,435 @@
+
+        obj_in_data = jsonable_encoder(data)
+        db_obj = model(**obj_in_data)
+
+        for child, rel_name in zip(children, children_rel_names, strict=True):
+            child_obj = self.add_insert(child, commit=True, id=None)
+            getattr(db_obj, rel_name).append(child_obj)
+
+        for key, value in data_table.items():
+            data_table_model = resolve_model(blueprint=bp, get_blueprint=self.get_blueprint, data_table_name=key)
+            data_table_objects = [data_table_model(data=item) for item in value]
+            getattr(db_obj, key).extend(data_table_objects)
+        if "type" in entity:
+            db_obj.type = entity["type"]
+        if "_id" in entity:
+            db_obj.id = entity["_id"]
+        if id:
+            db_obj.id = id
+        session.add(db_obj)
+        if commit:
+            session.commit()
+            session.refresh(db_obj)
+        return db_obj
+
+    def update(self, id: str, entity: dict):
+        self.add_table(entity["type"])
+        self.delete(id)
+        self.add_insert(entity)
+
+    def delete(self, id: str, table: str | None = None):
+        session = self.Session
+        metadata_obj = MetaData()
+        metadata_obj.reflect(bind=self.engine)
+        allowed_tables = [
+            i for i in metadata_obj.tables if not (i.endswith("_map") or i.endswith("_value") or i == "BP_Addresses")
+        ]
+        try:
+            for table in allowed_tables:
+                query = text(f"SELECT * FROM public.\"{table}\" WHERE id = '{id}';")  # noqa
+                result = self.Session.execute(query).first()
+                if result:
+                    query = text('SELECT "Address" FROM public."BP_Addresses" WHERE "Name" = :table').bindparams(
+                        table=table
+                    )
+                    address = session.execute(query).first()[0]
+                    if address:
+                        model, bp = self._resolve_blueprint(address)
+                        obj = self.Session.query(model).get(id)
+                        parent_mapping_tables = [i for i in metadata_obj.tables if (i.endswith(f"{table}_map"))]
+                        for j in parent_mapping_tables:
+                            query = text(f'delete FROM public."{j}" WHERE "{table}_id" = \'{id}\';')  # noqa
+                            self.Session.execute(query)
+                            self.Session.commit()
+                        self.Session.delete(obj)
+                        self.Session.flush()
+                        self.Session.commit()
+                        print(f"Successfully deleted {model.__name__}'{id}'")
+        except Exception as e:
+            print(f"An error occurred: {e}")
+        return jsonable_encoder(f"Could not find id:{id} in database")
+
+    def _resolve_blueprint(self, address):
+        bp = self.get_blueprint.get_blueprint(address).to_dict()
```

Comment:
```suggestion
        bp = self.get_blueprint.get_blueprint_with_extended_attributes(address).to_dict()
```
This will include attributes defined in other blueprints that the main blueprint "extends" from.